include/libcyphal/transport/can/media.hpp

```diff
@@ -0,0 +1,88 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef LIBCYPHAL_TRANSPORT_CAN_MEDIA_HPP_INCLUDED
+#define LIBCYPHAL_TRANSPORT_CAN_MEDIA_HPP_INCLUDED
+
+#include "libcyphal/types.hpp"
+#include "libcyphal/transport/errors.hpp"
+#include "libcyphal/transport/defines.hpp"
+
+namespace libcyphal
+{
+namespace transport
+{
+namespace can
+{
+
+using CanId = std::uint32_t;
+
+struct Filter final
+{
+    CanId id;
+    CanId mask;
+};
+using Filters = cetl::span<const Filter>;
+
+struct RxMetadata final
+{
+    TimePoint   timestamp;
+    CanId       can_id;
+    std::size_t payload_size;
+};
+
+class IMedia
+{
+public:
+    IMedia(IMedia&&)                 = delete;
+    IMedia(const IMedia&)            = delete;
+    IMedia& operator=(IMedia&&)      = delete;
+    IMedia& operator=(const IMedia&) = delete;
+
+    /// @brief Get the maximum transmission unit (MTU) of the CAN bus.
+    ///
+    /// This value may change arbitrarily at runtime. The transport implementation will query it before every
+    /// transmission on the port. This value has no effect on the reception pipeline as it can accept arbitrary MTU.
+    ///
+    CETL_NODISCARD virtual std::size_t getMtu() const noexcept = 0;
+
+    /// @brief Set the filters for the CAN bus.
+    ///
+    /// If there are fewer hardware filters available than requested, the configuration will be coalesced as described
+    /// in the Cyphal/CAN Specification. If zero filters are requested, all incoming traffic will be rejected.
+    /// While reconfiguration is in progress, incoming frames may be lost and/or unwanted frames may be received.
+    /// The lifetime of the filter array may end upon return (no references retained).
+    ///
+    /// @return Returns `true` on success, `false` in case of a low-level error (e.g., IO error).
+    ///
+    CETL_NODISCARD virtual bool setFilters(const Filters filters) noexcept = 0;
```

Comment:
>Why not MediaError here?

Ah yes, good point. @serges147 I think we need an expected here with appropriate error options: PlatformError (I think this is what Scott was referring to instead of MediaError as we don't have that error type), ArgumentError, CapacityError.

>Also, should we add a bool do_not_coalescence argument that means the set will fail if there aren't enough hardware filters when this is true?

No, but we could add `do_not_coalesce` :) If coalescence is disabled and there are not enough filters you get CapacityError.

Reply:
I will add `cetl::optional<MediaError>`.
`do_not_coalesce` will be addressed by issue #348

Reply:
What Scott called `MediaError` is currently named `PlatformError` in the design doc and the codebase. We need to either use `PlatformError` or rename it into `MediaError`.